### DISCUSSION TITLE
E13 (comm. error) on Maldives Hydrojet Pro 54173

### DISCUSSION THREAD

**Discussion by sergiopetruccelli** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#91] [Q&A]
Hi to all!
I have a Bestway Maldives Hydrojet Pro model 54173 (4 wire, Italy), I would add this remote control feature to it.

![image](https://user-images.githubusercontent.com/68513063/125170217-2c604d80-e1ae-11eb-859f-70d98990a328.png)

Circuit board build and ok, esp8266 flashed, webserver and mqtt works well.
But when I connect the "magic board" between CIO and DISP, after ten seconds I receive E13.
![image](https://user-images.githubusercontent.com/68513063/125170114-962c2780-e1ad-11eb-8039-c268115c36a8.png)
Pin declaration in BWC_82664w.cpp says this at default:
`void CIO::begin() {
  //Setup serial to CIO here
  //Using hardware UART/Serial on default pins (RX/TX)
  cio_serial.begin(9600, SWSERIAL_8N1, D6, D7, false, 64);
  cio_serial.setTimeout(50);
  dsp_serial.begin(9600, SWSERIAL_8N1, D1, D2, false, 64);
  dsp_serial.setTimeout(50);`
Not work as is.
In the schematic CIO pinouts says rx=D3 and tx=D2
![image](https://user-images.githubusercontent.com/68513063/125170564-cffe2d80-e1af-11eb-9d13-0007f4938230.png)

Edited, but still not working.
Hardware I think is ok, D2,D3,D6 and D7 pin gone up with simple sketch on 8266, when up I can read 5V on relative connector pin. 
There are other edit to made this fantastic project working on my 54713 Maldives Hydrojet?
Thanks in advance for you awesome job.

Sergio from Italy :)

**Comment by visualapproach** [comment]
Hi Italy!

Busy times now, but I plan to pay some attention to the 4-wire soon.

Cheers and good luck tonight

**Comment by visualapproach** [comment]
@sergiopetruccelli  Congrats to both Eurovision and football victory! 
Now, you are right about those pins. Amazing that anybody had this working. At least I think someone has. Anyway, I hope it's fixed in the latest update in the dev branch. I also added a simple program "serial bypass". If you have problems with E13 you could open that folder in pio, and upload. It does nothing but retransmitting the serial data. So my thoughts are that if you still get occasional E13 errors, it's hardware related.

**Reply by Lukey1979** [reply]
Hi @visualapproach , Just wondering how we go about uploading the "Serial Bypass" program to test to see if this fixes our E13 errors? We can see the folder in PIO but when we upload it, it completely overwrites the ESP8266 and we no longer have wifi access to it.

**Reply by visualapproach** [reply]
@Lukey1979 it seems you uploaded it alright. If it works without E13 your hardware is ok. If not there is probably a hardware issue. Or the pump is really picky about the latency between the ports. Or the planets are not in the right configuration.
All that the program is supposed to do, is to copy data between the ports without screwing about.

**Comment by Lukey1979** [comment]
Hi @sergiopetruccelli Did you end up fixing the E13 errors? if you did, what was the fix?

**Comment by Bishbosh02** [comment]
Hi I have a 4 wire old stile palm springs and my pinouts were different DSP (1=5V,2=TX,3=RX,4=GND)  CIO (1=5V,2=RX,3=TX,4=GND) also I used an esp32 just upgrade pump to new style 6 wire so that's why I am now trying this solution.

**Reply by visualapproach** [reply]
Hi, the ESP32 won't work on 6-wire. Only 8266.

**Reply by Bishbosh02** [reply]
Ok the esp32 was for my old 4 wire controller running my own code and PCB, now that I have a 6 wire pump unit I plan to use your design (why use an esp32 when a 8266 will do the job :) )   I was going to add power monitor using ADS1115 but your already using D1 and D1 so will test how accurate the ADC or try using different pins etc.

Have the 8266 running on the bench and have order an DSP and CIO from ebay so that I can test it before striping my new pump down, my old design used Pushsafer so I could push notifications to my phone and Espalexa so that I could turn on the blower from alexa....

**Reply by Bishbosh02** [reply]
D1 and D2 :(

**Reply by visualapproach** [reply]
Hi there,
You could use other pins with some limitations. Rx/Tx could be used if you don't use serial. Not sure the CPU is fast enough to cope with fast traffic i2c while bitbanging the hot tub but you can always try.

**Comment by visualapproach** [comment]
The v2.1.4 of the code may solve this E13 problem.

**Comment by Xamude** [comment]
Have you get it to work ?
I tested some things but it dosen´t work at the moment:(

Measure 6,9V on BLK - RED  is this normal ? ( Red = GND , BLK = "5"V)

Powered ESP with Powerbank and connect the DSP -> Web IF says connected.
If i connect now the CIO with only Data the ESP only blinks .

Powered ESP by CIO without DATA an connect the DSP -> ESP blinks.

Tryed to switch DATA Cabels on CIO input without DSP connected but never get DSP active on Webpage.

In Model.h i use #define NO54173  but on the Webpage are no Jet´s visible. 
I´m not sure if this part of my issue.

Not tested at the moment is Cap, Resistors an short Wires. 
Use development branch from 22.4 

I relay hope to get this to work 😅
Thanks in advance for your help

**Reply by visualapproach** [reply]
Hi, I'm not sure what's normal or not but I think I measured 5-ish volts. Measure with load on (connected to display or esp) and see if it approaches 5V. The ESP can tolerate 6.9V but the TXS0108E states 6.5 max on high side.

**Reply by visualapproach** [reply]
You could test the passthrough code (found in separate folder) to see if the pump works as normal. If so it should behave like the esp was not there.

**Reply by visualapproach** [reply]
@Xamude the hydrojets button was broken. Fix in dev branch 2022-05-18. You need to upload index.html and websockets.js in addition to the new code.

**Reply by Xamude** [reply]
Dev branch looks good.
Had not much time today to test.
It´s weird with the Voltage had 6,3V , but i measured again after some tests with resets an had 6,9V.
Pass through Code tested on a different MCU an all connected to the pump -> E 13
Other test with Powerbank and GND connected an No Power from Pump -> Display does nothing

I thing Problem is be between Pump and MCU
Maybe a Problem with the TXS0108E ? I read something in an other Post that they have problems. 
(What was the Type of the other ? I will order a 4 Channel for testing )

**Reply by visualapproach** [reply]
https://www.electrokit.com/uploads/productimage/41014/41014895-2.jpg

There is 8 ch modules too

**Reply by Xamude** [reply]
Tested now with new converter , the Connection MCU -> Pump works :)
But i still get E13 Errors.
Start Filter Pump over Display ~ 1 Minute -> E13
Use Web If after start Up -> E 13
Use Web if -> Take Control -> no E 13 (but not enough tested at the Moment )
 
Test now if Take Control ends in E 13 and test pass truth  Code only.
Maybe can a bad WiFi cause this ?

Now i´m going out for more testing :D

**Reply by visualapproach** [reply]
Yeah it seems bad Wi-Fi can make web sockets lock up long enough to trigger the hardware watchdog. Check reboot reason and if you know how, try setting web sockets timeout to 1000 or something low.

**Reply by Xamude** [reply]
Hello, had no time the last week´s to test something :(
I installed a Wifi Router 1,5 Meters away from the Pool the exclude Wifi issues.
Had some test runs with the Passthrue Code this works for some hours without issues.
At the moment i had run the normal Code (Firmware: 4W_2022-05-18) since 20 Minutes without E13 Erros,
but i can´t control it via Web If, only on display. When i use "Take Control" then i can control it over Web If.

Hope i had more time to test it in the next weeks 

EDIT:
find the Answer to Web if or DSP Control:
[https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/discussions/296](url) 
how should i tell that my wife ?😂

1 1/2 hour late I get many e 13 in a row (in god Mode)

**Reply by visualapproach** [reply]
> how should i tell that my wife ?

Someone (I forgot who) posted a picture where he had attached a nail, point up, to the buttons 😄 

I can't tell what to do to the errors at the moment. When I get time I'll see if I can do anything codewise.

**Reply by Xamude** [reply]
![image](https://user-images.githubusercontent.com/66908227/173226900-b58e516d-03ff-464a-8776-23a212cbd5e0.png)
I think that this help´s to exclude Wifi errors  :D

`9:30 Uhr Boot -> Take Control -> Star Pump
9:45 -> Stat Bubbles for 10 Seconds
	 -> Start Hydro for 10 Seconds
	 -> Start Pump again
	 
RSSI: is between -26 & -30

10:00 Noticed the Pump stoped working
		looking in Web If Take Control and Pump was off
		Aktivated booth ( no E13 until now )
		
		
10:15  noticed E13 
		Take Control and Pump was off
		Pump Started and Stoped in a loop 

		MQTT: undefined
		CIO TX: Active
		DSP TX: Active
		Firmware: 4W_2022-05-18
		
10:18 Cut Power and started again 

10:30 E13 Error 
		Reconnect the DSP -> E 13
		
10:35	Cut Power and start again
		Aktivated Take Control on Web if
		Started Pump over MQTT
		
11:20 Had to stop testing :/
`
Maybe it trigger an E13 when i had the website open and the notebook goes in sleep mode ?
As every time i will keep on testing and report 😅

**Reply by visualapproach** [reply]
I haven't solved anything, but added loop time info in the dev branch. If you want to try, you can log the LOOPMAX value, found in "OTHER" topic in both websockets and mqtt. I guess it will be a high value after an error E13.

**Reply by Xamude** [reply]
Today i have enough time for tests.
Had one E 13 at startup bot now it works for hours without E 13.

Now 2 times happened that the ESP "forgot" what it had to do:
God mode, Filter and Heater went from true/1 to false/0
Can turn all on again over MQTT / Web if without E 13

Log:
`{"CONTENT":"STATES","TIME":1655371582,"LCK":0,"PWR":1,"UNT":1,"AIR":0,"GRN":0,"RED":1,"FLT":1,"TGT":33,"TMP":31,"CH1":32,"CH2":32,"CH3":32,"JET":0,"ERR":0,"GOD":true}
debug.html:45 {"CONTENT":"TIMES","TIME":1655371582,"CLTIME":1655330034,"FTIME":1655330033,"UPTIME":13990,"PUMPTIME":9340,"HEATINGTIME":6649,"AIRTIME":37,"JETTIME":30,"KWH":3.632999897,"COST":0.980910003,"FINT":30,"CLINT":14,"TTTT":0}
debug.html:45 {"CONTENT":"OTHER","MQTT":0,"CIOTX":true,"DSPTX":true,"HASJETS":true,"RSSI":-27,"IP":"192.168.170.143","SSID":"meins","FW":"4W_2022-06-12","MODEL":"NO54173","LOOPMAX":1219,"LOOPMIN":0}
debug.html:45 {"CONTENT":"STATES","TIME":1655371583,"LCK":0,"PWR":1,"UNT":1,"AIR":0,"GRN":0,"RED":1,"FLT":1,"TGT":33,"TMP":31,"CH1":32,"CH2":32,"CH3":32,"JET":0,"ERR":0,"GOD":true}
debug.html:45 {"CONTENT":"TIMES","TIME":1655371583,"CLTIME":1655330034,"FTIME":1655330033,"UPTIME":13991,"PUMPTIME":9341,"HEATINGTIME":6649,"AIRTIME":37,"JETTIME":30,"KWH":3.632999897,"COST":0.980910003,"FINT":30,"CLINT":14,"TTTT":0}
debug.html:45 {"CONTENT":"OTHER","MQTT":0,"CIOTX":true,"DSPTX":true,"HASJETS":true,"RSSI":-28,"IP":"192.168.170.143","SSID":"meins","FW":"4W_2022-06-12","MODEL":"NO54173","LOOPMAX":1219,"LOOPMIN":0}
debug.html:45 {"CONTENT":"STATES","TIME":1655371583,"LCK":0,"PWR":1,"UNT":1,"AIR":0,"GRN":0,"RED":1,"FLT":1,"TGT":33,"TMP":31,"CH1":32,"CH2":32,"CH3":32,"JET":0,"ERR":0,"GOD":true}
debug.html:45 {"CONTENT":"TIMES","TIME":1655371583,"CLTIME":1655330034,"FTIME":1655330033,"UPTIME":13992,"PUMPTIME":9341,"HEATINGTIME":6650,"AIRTIME":37,"JETTIME":30,"KWH":3.632999897,"COST":0.980910003,"FINT":30,"CLINT":14,"TTTT":0}
debug.html:45 {"CONTENT":"OTHER","MQTT":0,"CIOTX":true,"DSPTX":true,"HASJETS":true,"RSSI":-27,"IP":"192.168.170.143","SSID":"meins","FW":"4W_2022-06-12","MODEL":"NO54173","LOOPMAX":1219,"LOOPMIN":0}
debug.html:45 {"CONTENT":"STATES","TIME":1655371584,"LCK":0,"PWR":1,"UNT":1,"AIR":0,"GRN":0,"RED":1,"FLT":1,"TGT":33,"TMP":31,"CH1":32,"CH2":32,"CH3":32,"JET":0,"ERR":0,"GOD":true}
debug.html:45 {"CONTENT":"TIMES","TIME":1655371584,"CLTIME":1655330034,"FTIME":1655330033,"UPTIME":13993,"PUMPTIME":9342,"HEATINGTIME":6651,"AIRTIME":37,"JETTIME":30,"KWH":3.634000063,"COST":0.981180072,"FINT":30,"CLINT":14,"TTTT":0}
debug.html:45 {"CONTENT":"OTHER","MQTT":0,"CIOTX":true,"DSPTX":true,"HASJETS":true,"RSSI":-28,"IP":"192.168.170.143","SSID":"meins","FW":"4W_2022-06-12","MODEL":"NO54173","LOOPMAX":1219,"LOOPMIN":0}
debug.html:45 {"CONTENT":"STATES","TIME":1655371585,"LCK":0,"PWR":1,"UNT":1,"AIR":0,"GRN":0,"RED":1,"FLT":1,"TGT":33,"TMP":63,"CH1":69,"CH2":60,"CH3":56,"JET":0,"ERR":128,"GOD":false}
debug.html:45 {"CONTENT":"TIMES","TIME":1655371585,"CLTIME":1655330034,"FTIME":1655330033,"UPTIME":13994,"PUMPTIME":9343,"HEATINGTIME":6652,"AIRTIME":37,"JETTIME":30,"KWH":3.634000063,"COST":0.981180072,"FINT":30,"CLINT":14,"TTTT":0}
debug.html:45 {"CONTENT":"OTHER","MQTT":0,"CIOTX":true,"DSPTX":true,"HASJETS":true,"RSSI":-28,"IP":"192.168.170.143","SSID":"meins","FW":"4W_2022-06-12","MODEL":"NO54173","LOOPMAX":1219,"LOOPMIN":0}
debug.html:45 {"CONTENT":"STATES","TIME":1655371586,"LCK":0,"PWR":1,"UNT":1,"AIR":0,"GRN":2,"RED":0,"FLT":0,"TGT":33,"TMP":31,"CH1":32,"CH2":32,"CH3":32,"JET":0,"ERR":0,"GOD":false}
debug.html:45 {"CONTENT":"TIMES","TIME":1655371586,"CLTIME":1655330034,"FTIME":1655330033,"UPTIME":13995,"PUMPTIME":9344,"HEATINGTIME":6652,"AIRTIME":37,"JETTIME":30,"KWH":3.634000063,"COST":0.981180072,"FINT":30,"CLINT":14,"TTTT":0}
debug.html:45 {"CONTENT":"OTHER","MQTT":0,"CIOTX":true,"DSPTX":true,"HASJETS":true,"RSSI":-28,"IP":"192.168.170.143","SSID":"meins","FW":"4W_2022-06-12","MODEL":"NO54173","LOOPMAX":1219,"LOOPMIN":0}
debug.html:45 {"CONTENT":"STATES","TIME":1655371587,"LCK":0,"PWR":1,"UNT":1,"AIR":0,"GRN":2,"RED":0,"FLT":0,"TGT":33,"TMP":31,"CH1":32,"CH2":32,"CH3":32,"JET":0,"ERR":0,"GOD":false}
`

**Reply by Xamude** [reply]
New Error : 
En8 
Reconnected the DSP and Set Godmode , FLT and Heater true/1


No E13 Until Now :)

`Last boot: 16.6.2022, 08:56:09 Power On
{"CONTENT":"STATES","TIME":1655375419,"LCK":0,"PWR":1,"UNT":1,"AIR":0,"GRN":0,"RED":1,"FLT":1,"TGT":33,"TMP":32,"CH1":32,"CH2":32,"CH3":32,"JET":0,"ERR":0,"GOD":true}
debug.html:45 {"CONTENT":"TIMES","TIME":1655375419,"CLTIME":1655330034,"FTIME":1655330033,"UPTIME":17828,"PUMPTIME":12934,"HEATINGTIME":9197,"AIRTIME":41,"JETTIME":110,"KWH":5.038000107,"COST":1.360260129,"FINT":30,"CLINT":14,"TTTT":0}
debug.html:45 {"CONTENT":"OTHER","MQTT":0,"CIOTX":true,"DSPTX":true,"HASJETS":true,"RSSI":-28,"IP":"192.168.170.143","SSID":"meins","FW":"4W_2022-06-12","MODEL":"NO54173","LOOPMAX":1219,"LOOPMIN":0}
debug.html:45 {"CONTENT":"STATES","TIME":1655375420,"LCK":0,"PWR":1,"UNT":1,"AIR":0,"GRN":0,"RED":1,"FLT":1,"TGT":33,"TMP":32,"CH1":32,"CH2":32,"CH3":32,"JET":0,"ERR":0,"GOD":true}
debug.html:45 {"CONTENT":"TIMES","TIME":1655375420,"CLTIME":1655330034,"FTIME":1655330033,"UPTIME":17828,"PUMPTIME":12934,"HEATINGTIME":9197,"AIRTIME":41,"JETTIME":110,"KWH":5.038000107,"COST":1.360260129,"FINT":30,"CLINT":14,"TTTT":0}
debug.html:45 {"CONTENT":"OTHER","MQTT":0,"CIOTX":true,"DSPTX":true,"HASJETS":true,"RSSI":-28,"IP":"192.168.170.143","SSID":"meins","FW":"4W_2022-06-12","MODEL":"NO54173","LOOPMAX":1219,"LOOPMIN":0}
debug.html:45 {"CONTENT":"STATES","TIME":1655375421,"LCK":0,"PWR":1,"UNT":1,"AIR":0,"GRN":0,"RED":1,"FLT":1,"TGT":33,"TMP":32,"CH1":32,"CH2":32,"CH3":32,"JET":0,"ERR":0,"GOD":true}
debug.html:45 {"CONTENT":"TIMES","TIME":1655375421,"CLTIME":1655330034,"FTIME":1655330033,"UPTIME":17829,"PUMPTIME":12935,"HEATINGTIME":9198,"AIRTIME":41,"JETTIME":110,"KWH":5.039000034,"COST":1.360530019,"FINT":30,"CLINT":14,"TTTT":0}
debug.html:45 {"CONTENT":"OTHER","MQTT":0,"CIOTX":true,"DSPTX":true,"HASJETS":true,"RSSI":-28,"IP":"192.168.170.143","SSID":"meins","FW":"4W_2022-06-12","MODEL":"NO54173","LOOPMAX":1219,"LOOPMIN":0}
debug.html:45 {"CONTENT":"STATES","TIME":1655375421,"LCK":0,"PWR":1,"UNT":1,"AIR":0,"GRN":0,"RED":1,"FLT":1,"TGT":33,"TMP":32,"CH1":32,"CH2":32,"CH3":32,"JET":0,"ERR":0,"GOD":true}
debug.html:45 {"CONTENT":"TIMES","TIME":1655375421,"CLTIME":1655330034,"FTIME":1655330033,"UPTIME":17830,"PUMPTIME":12936,"HEATINGTIME":9199,"AIRTIME":41,"JETTIME":110,"KWH":5.039999962,"COST":1.360800028,"FINT":30,"CLINT":14,"TTTT":0}
debug.html:45 {"CONTENT":"OTHER","MQTT":0,"CIOTX":true,"DSPTX":true,"HASJETS":true,"RSSI":-28,"IP":"192.168.170.143","SSID":"meins","FW":"4W_2022-06-12","MODEL":"NO54173","LOOPMAX":1219,"LOOPMIN":0}
debug.html:45 {"CONTENT":"STATES","TIME":1655375423,"LCK":0,"PWR":1,"UNT":1,"AIR":0,"GRN":0,"RED":1,"FLT":1,"TGT":33,"TMP":32,"CH1":32,"CH2":32,"CH3":32,"JET":0,"ERR":0,"GOD":true}
debug.html:45 {"CONTENT":"TIMES","TIME":1655375423,"CLTIME":1655330034,"FTIME":1655330033,"UPTIME":17832,"PUMPTIME":12938,"HEATINGTIME":9201,"AIRTIME":41,"JETTIME":110,"KWH":5.040999889,"COST":1.361070037,"FINT":30,"CLINT":14,"TTTT":0}
debug.html:45 {"CONTENT":"OTHER","MQTT":0,"CIOTX":true,"DSPTX":true,"HASJETS":true,"RSSI":-28,"IP":"192.168.170.143","SSID":"meins","FW":"4W_2022-06-12","MODEL":"NO54173","LOOPMAX":1219,"LOOPMIN":0}
debug.html:45 {"CONTENT":"STATES","TIME":1655375424,"LCK":0,"PWR":1,"UNT":1,"AIR":0,"GRN":0,"RED":1,"FLT":1,"TGT":33,"TMP":32,"CH1":69,"CH2":60,"CH3":56,"JET":0,"ERR":128,"GOD":false}
debug.html:45 {"CONTENT":"TIMES","TIME":1655375424,"CLTIME":1655330034,"FTIME":1655330033,"UPTIME":17832,"PUMPTIME":12938,"HEATINGTIME":9201,"AIRTIME":41,"JETTIME":110,"KWH":5.040999889,"COST":1.361070037,"FINT":30,"CLINT":14,"TTTT":0}
debug.html:45 {"CONTENT":"OTHER","MQTT":0,"CIOTX":true,"DSPTX":true,"HASJETS":true,"RSSI":-27,"IP":"192.168.170.143","SSID":"meins","FW":"4W_2022-06-12","MODEL":"NO54173","LOOPMAX":1219,"LOOPMIN":0}
debug.html:45 {"CONTENT":"STATES","TIME":1655375424,"LCK":0,"PWR":1,"UNT":1,"AIR":0,"GRN":2,"RED":0,"FLT":0,"TGT":33,"TMP":32,"CH1":32,"CH2":32,"CH3":32,"JET":0,"ERR":0,"GOD":false}
debug.html:45 {"CONTENT":"TIMES","TIME":1655375424,"CLTIME":1655330034,"FTIME":1655330033,"UPTIME":17832,"PUMPTIME":12938,"HEATINGTIME":9201,"AIRTIME":41,"JETTIME":110,"KWH":5.040999889,"COST":1.361070037,"FINT":30,"CLINT":14,"TTTT":0}
debug.html:45 {"CONTENT":"OTHER","MQTT":0,"CIOTX":true,"DSPTX":true,"HASJETS":true,"RSSI":-27,"IP":"192.168.170.143","SSID":"meins","FW":"4W_2022-06-12","MODEL":"NO54173","LOOPMAX":1219,"LOOPMIN":0}
debug.html:45 {"CONTENT":"STATES","TIME":1655375425,"LCK":0,"PWR":1,"UNT":1,"AIR":0,"GRN":2,"RED":0,"FLT":0,"TGT":33,"TMP":32,"CH1":32,"CH2":32,"CH3":32,"JET":0,"ERR":0,"GOD":false}
debug.html:45 {"CONTENT":"TIMES","TIME":1655375425,"CLTIME":1655330034,"FTIME":1655330033,"UPTIME":17834,"PUMPTIME":12938,"HEATINGTIME":9201,"AIRTIME":41,"JETTIME":110,"KWH":5.040999889,"COST":1.361070037,"FINT":30,"CLINT":14,"TTTT":0}
debug.html:45 {"CONTENT":"OTHER","MQTT":0,"CIOTX":true,"DSPTX":true,"HASJETS":true,"RSSI":-28,"IP":"192.168.170.143","SSID":"meins","FW":"4W_2022-06-12","MODEL":"NO54173","LOOPMAX":1219,"LOOPMIN":0}
debug.html:45 {"CONTENT":"STATES","TIME":1655375426,"LCK":0,"PWR":1,"UNT":1,"AIR":0,"GRN":2,"RED":0,"FLT":0,"TGT":33,"TMP":32,"CH1":32,"CH2":32,"CH3":32,"JET":0,"ERR":0,"GOD":false}
debug.html:45 {"CONTENT":"TIMES","TIME":1655375426,"CLTIME":1655330034,"FTIME":1655330033,"UPTIME":17834,"PUMPTIME":12938,"HEATINGTIME":9201,"AIRTIME":41,"JETTIME":110,"KWH":5.040999889,"COST":1.361070037,"FINT":30,"CLINT":14,"TTTT":0}
debug.html:45 {"CONTENT":"OTHER","MQTT":0,"CIOTX":true,"DSPTX":true,"HASJETS":true,"RSSI":-27,"IP":"192.168.170.143","SSID":"meins","FW":"4W_2022-06-12","MODEL":"NO54173","LOOPMAX":1219,"LOOPMIN":0}
debug.html:45 {"CONTENT":"STATES","TIME":1655375426,"LCK":0,"PWR":1,"UNT":1,"AIR":0,"GRN":2,"RED":0,"FLT":0,"TGT":33,"TMP":32,"CH1":32,"CH2":32,"CH3":32,"JET":0,"ERR":0,"GOD":false}
debug.html:45 {"CONTENT":"TIMES","TIME":1655375426,"CLTIME":1655330034,"FTIME":1655330033,"UPTIME":17834,"PUMPTIME":12938,"HEATINGTIME":9201,"AIRTIME":41,"JETTIME":110,"KWH":5.040999889,"COST":1.361070037,"FINT":30,"CLINT":14,"TTTT":0}
debug.html:45 {"CONTENT":"OTHER","MQTT":0,"CIOTX":true,"DSPTX":true,"HASJETS":true,"RSSI":-27,"IP":"192.168.170.143","SSID":"meins","FW":"4W_2022-06-12","MODEL":"NO54173","LOOPMAX":1219,"LOOPMIN":0}
debug.html:45 {"CONTENT":"STATES","TIME":1655375427,"LCK":0,"PWR":1,"UNT":1,"AIR":0,"GRN":2,"RED":0,"FLT":0,"TGT":33,"TMP":32,"CH1":32,"CH2":32,"CH3":32,"JET":0,"ERR":0,"GOD":false}
debug.html:45 {"CONTENT":"TIMES","TIME":1655375427,"CLTIME":1655330034,"FTIME":1655330033,"UPTIME":17836,"PUMPTIME":12938,"HEATINGTIME":9201,"AIRTIME":41,"JETTIME":110,"KWH":5.040999889,"COST":1.361070037,"FINT":30,"CLINT":14,"TTTT":0}
debug.html:45 {"CONTENT":"OTHER","MQTT":0,"CIOTX":true,"DSPTX":true,"HASJETS":true,"RSSI":-28,"IP":"192.168.170.143","SSID":"meins","FW":"4W_2022-06-12","MODEL":"NO54173","LOOPMAX":1219,"LOOPMIN":0}
`

EDIT:
Had an E 13 Error at 16:43 ( in log file it is 14:43)
Pool was in Manual Mode but i had the debug page open in Chrome and closed the Notebook that it goes in standby,
in this Moment the E 13 appears.
The Loop Max shows 15798 in my Loop Max log and 30011 in the File log. 

[Raum_Garten_Whirlpool_message (2).csv](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/files/8921844/Raum_Garten_Whirlpool_message.2.csv)
[Raum_Garten_Whirlpool_other (1).csv](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/files/8921845/Raum_Garten_Whirlpool_other.1.csv)

Reconnect the Display and no more Error until 21 o´clock

**Reply by visualapproach** [reply]
> Had an E 13 Error at 16:43 ( in log file it is 14:43)
Pool was in Manual Mode but i had the debug page open in Chrome and closed the Notebook that it goes in standby,
in this Moment the E 13 appears.
The Loop Max shows 15798 in my Loop Max log and 30011 in the File log.

Good to know. Then we "just" need to prevent wifi from blocking the loop.

**Reply by Xamude** [reply]
> 
> Good to know. Then we "just" need to prevent wifi from blocking the loop.

:D 
i think can live with 1 E13 in the day,when there is no fix .
Will try do add a Switch in DSP  5V Line to reset the E13 ,if this work´s, it shut be possible to add it via a Relay.

Interesting is that the E13 mostly appears between 16 & 17 o´clock
Maybe you can add a logfile that stores the data from "message" and "other"?
The logging over MQTT is not  satisfactory .

Thanks for all your work to help to solve my Problems  👍

**Reply by visualapproach** [reply]
@Xamude check out the new code in development branch. I pulled the fork (can you eat pulled fork??) from #304. You are welcome to test it and see if it helps with your E13. Thank you @cobaltfish

**Reply by Xamude** [reply]
Good Evening,
Tested the Firmware: 4W_2022-06-27 without E 13 Errors an 3 day´s that´s great :) 
(test only on the weekend so i can reset the beep )

Problem is now that i get an En8 error and beep :(

and the other one is that the esp stops good mode and the pump..
had this ~ 7 times today 

EDIT:
other interesting thing: God & Pump active, when i open the WebIf God/Pump goes off

**Reply by visualapproach** [reply]
Is the error shown on the pump display? 'En8'?

**Reply by visualapproach** [reply]
@Xamude I made a very small update to the dev branch now. Before: if cio send a message with an error flag AND a bad checksum, we disarmed godmode. Now: we don't care about the message.

**Comment by Xamude** [comment]
![IMG_20220709_175620](https://user-images.githubusercontent.com/66908227/178139147-9bad52df-46f3-44da-83ac-4e83cbb255eb.jpg)
Yes it's on the Display.
Will update later.
Seem's we are on the right way to get it to work :)

**Reply by Xamude** [reply]
Sorry for the late response, had not much time to test it.
in 4 Test Day´s i got 3 En8 Errors
and 1day  E01 
but still no E13 :)

**Reply by visualapproach** [reply]
I don't know if it applies here, but @SigmaPic found out in another thread #330 that pins D3/4 was no good as inputs. I wonder if it would help to use other pins here too.
In that case you should move the wires going to d3 and d4 to for instance d5 and d6. And then edit pinout accordingly in the code
